src/Mod/Arch/ArchRoof.py

```diff
@@ -683,79 +842,68 @@ def getSubVolume(self,obj):
                         return self.sub
         return None
 
-    def computeAreas(self,obj):
-
-        "computes border and ridge roof edges length"
-
-        if hasattr(obj,"RidgeLength") and hasattr(obj,"BorderLength"):
+    def computeAreas(self, obj):
+        '''computes border and ridge roof edges length'''
+        if hasattr(obj, "RidgeLength") and hasattr(obj, "BorderLength"):
             rl = 0
             bl = 0
             rn = 0
             bn = 0
-            import Part,math
             if obj.Shape:
                 if obj.Shape.Faces:
-                    fset = []
-                    for f in obj.Shape.Faces:
-                        if f.normalAt(0,0).getAngle(FreeCAD.Vector(0,0,1)) < math.pi/2:
-                            fset.append(f)
-                    if fset:
+                    faceLst = []
+                    for face in obj.Shape.Faces:
+                        if face.normalAt(0, 0).getAngle(Vector(0.0, 0.0, 1.0)) < math.pi / 2.0:
+                            faceLst.append(face)
+                    if faceLst:
                         try:
-                            shell = Part.Shell(fset)
+                            shell = Part.Shell(faceLst)
```

Comment:
```py
        faces.append(face)

if faces:
    try:
        shell = Part.Shell(faces)
    except Part.OCCError:
        pass
```

By the way, we should try catching a specific exception, like `Part.OCCError`, and not just any exception.

Reply:
I do not know enough about this.

Reply:
Usually when a `Part` method fails to create a `Shape` it throws an exception which is very often `Part.OCCError`. So we just catch this specific exception, instead of using a generic `except`.